src/m365/teams/commands/meeting/meeting-add.ts

```diff
@@ -0,0 +1,216 @@
+          return `'${args.options.startTime}' is not a valid ISO date string for startTime.`;
+        }
+
+        if (args.options.endTime && !validation.isValidISODateTime(args.options.endTime)) {
+          return `'${args.options.endTime}' is not a valid ISO date string for endTime.`;
+        }
+
+        if (args.options.startTime && args.options.endTime && new Date(args.options.startTime) >= new Date(args.options.endTime)) {
+          return 'The startTime value must be before endTime.';
+        }
+
+        if (args.options.endTime && !args.options.startTime && new Date() >= new Date(args.options.endTime)) {
+          return 'When only the endTime is specified, it needs to be after the current time.';
+        }
+
+        if (args.options.participantUserNames) {
+
+          if (args.options.participantUserNames.indexOf(',') === -1 && !validation.isValidUserPrincipalName(args.options.participantUserNames)) {
+            return `${args.options.participantUserNames} contains invalid UPN.`;
+          }
+
+          const participants = args.options.participantUserNames.trim().toLowerCase().split(',').filter(e => e && e !== '');
+
+          if (!participants || participants.length === 0 || participants.some(e => !validation.isValidUserPrincipalName(e))) {
+            return `${args.options.participantUserNames} contains one or more invalid UPN.`;
+          }
+        }
+
+        if (args.options.organizerEmail && !validation.isValidUserPrincipalName(args.options.organizerEmail)) {
+          return `'${args.options.organizerEmail}' is not a valid email for organizerEmail.`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  /**
+   * Executes the command
+   * @param logger Logger instance
+   * @param args Command arguments
+   */
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const isAppOnlyAccessToken = accessToken.isAppOnlyAccessToken(auth.service.accessTokens[this.resource].accessToken)!;
+
+      if (isAppOnlyAccessToken && !args.options.organizerEmail) {
+        throw `The option 'organizerEmail' is required when creating a meeting using app only permissions`;
+      }
+
+      if (!isAppOnlyAccessToken && args.options.organizerEmail) {
+        throw `The option 'organizerEmail' is not supported when creating a meeting using delegated permissions`;
+      }
+
+      let graphBaseUrl = `${this.resource}/v1.0/`;
+
+      if (args.options.organizerEmail) {
+        const organizerId = await aadUser.getUserIdByEmail(args.options.organizerEmail);
+        graphBaseUrl = `${graphBaseUrl}users/${organizerId}`;
+      }
+      else {
+        graphBaseUrl = `${graphBaseUrl}me`;
+      }
+
+      const meeting = await this.createMeeting(logger, graphBaseUrl, args.options);
+      await logger.log(meeting);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+  /**
```

Comment:
Let's add an extra white line above this method

Reply:
Done.